Five min quickstart

eysi09 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first pull request, please read our contributor guidelines in the https://github.com/garden-io/garden/blob/main/CONTRIBUTING.md file.
2. Please label this pull request according to what type of issue you are addressing (see "What type of PR is this?" below)
3. Ensure you have added or run the appropriate tests for your PR.
4. If the PR is unfinished, add `WIP:` at the beginning of the title or use the Github Draft PR feature.
5. Please add at least two reviewers to the PR. Currently active maintainers are: @edvald, @thsig, @eysi09, @Orzelius and @vvagaytsev.
-->

**What this PR does / why we need it**:

A new "5 Minute Quickstart" guide for our docs. 

This is part of the larger docs restructure we're making where we:

- Add a single "plugin-wide" quickstart guide that uses the `local-kubernetes` plugin. This is that guide.
- Restructure the docs such that we group most content by plugins (i.e Kubernetes / Terraform / Exec etc).
- Add separate getting started guides for all the plugins (those mostly require more than 5 minutes).

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

- The example project is still in a private repo. I'll make it public once I've cleaned it up some more and gotten feedback.
- The funky markdown syntax is for displaying tables with selectable tabs on Gitbook ([like these](https://docs.gitbook.com/tour/editor/blocks/tabs#example)).
- The example is interactive, please give it a spin.

thsig commented on docs/getting-started/5-min-quickstart.md:
```diff
@@ -0,0 +1,146 @@
+
+* Git (v2.14 or newer)
+* rsync (v3.1.0 or newer)
+
```
Maybe add something like:
> Run `git version` and `rsync --version` to verify that a recent enough version of Git and rsync is installed.

thsig commented on docs/getting-started/5-min-quickstart.md:
````diff
@@ -0,0 +1,146 @@
+
+{% tab title="macOS (Homebrew)" %}
+```sh
+brew tap garden-io/garden
````
Just a thought: How about changing this to
```sh
brew tap garden-io/garden && brew install garden-cli
```
so that it's a one-liner execution in the shell?

thsig commented on docs/getting-started/5-min-quickstart.md:
```diff
@@ -0,0 +1,146 @@
+
+In this guide we're using local Kubernetes since that's usually the fastest way to get started.
+
+For real world projects we recommend using a remote Kubernetes cluster since that comes with various benefits such as shared caches and, well, the fact that you don't need to run K8s on your laptop.
```
```suggestion
For real world projects we recommend using a remote Kubernetes cluster since that comes with various benefits such as shared caches and, well, doesn't need you to run K8s on your laptop!
```

thsig commented on docs/getting-started/5-min-quickstart.md:
```diff
@@ -0,0 +1,146 @@
+1. From the Docker Dashboard, select the **Preferences** icon.
+2. Select **Kubernetes** from the left sidebar.
+3. Next to **Enable Kubernetes**, select the checkbox.
+3. Select **Apply & Restart** to save the settings and then click Install to confirm. This instantiates images required to run the Kubernetes server as containers, and installs kubectl on your machine.
```
```suggestion
3. Select **Apply & Restart** to save the settings and then click Install to confirm. This instantiates the images required to run the Kubernetes server as containers, and installs kubectl on your machine.
```

thsig reviewed (commented):
Good stuff! Added a few suggestions there.

eysi09 commented on docs/getting-started/5-min-quickstart.md:
```diff
@@ -0,0 +1,146 @@
+In the guide we'll:
+
+* Install Garden
+* Install local Kubernetes
```
Haha! No strong opinions on my end.

eysi09 replied:
Actually, I'm leaning towards the current version. It flows more naturally in this context.

eysi09 commented:
Thanks for the feedback! I've addressed the comments and also added a "Next Steps" section. 

Note that this is only a temp version of that and we'll update when we have better "next steps" guides. 

I'd also be very curious to know what you thought of the example project. 

That's basically the star of the show. It could definitely do with better styling and improved text, but I wanted to hold off on taking that too far until I get feedback on the actual concept. 

I'd rather not spend time wrangling CSS if people hate the concept either way.

vvagaytsev commented:
Is it good to go @thsig @Orzelius? I've resolved the discussions that were addressed in https://github.com/garden-io/garden/pull/3307/commits/73b2af8142905b23daf12a71480d2e64a550eaca.

We still have an open one - about the "local Kubernetes" :)

Orzelius reviewed (dismissed):
good from me

vvagaytsev commented:
Ah, [check-docs step](https://app.circleci.com/pipelines/github/garden-io/garden/13514/workflows/2155144a-4fff-4a21-a3d5-196d77b812a6/jobs/224980) failed.

@eysi09 should the new guide be in the "Basics" section or in the "Getting started"?

eysi09 commented:
> Ah, [check-docs step](https://app.circleci.com/pipelines/github/garden-io/garden/13514/workflows/2155144a-4fff-4a21-a3d5-196d77b812a6/jobs/224980) failed.
> 
> @eysi09 should the new guide be in the "Basics" section or in the "Getting started"?

Basic for now. But we'll restructure the whole thing.

vvagaytsev commented:
I've pushed https://github.com/garden-io/garden/pull/3307 to fix the `check-docs` CI step. @Orzelius can you review this again, please?

vvagaytsev commented:
Some links were broken in the new readme guide. Now it's fixed in https://github.com/garden-io/garden/pull/3307/commits/1c88e9ca5f675e37f7f3504994fe4a608acd9108.

The `check-docs` step passed! 🎉 :)
